commands/install.bee.inc

```diff
@@ -84,17 +96,40 @@ function install_bee_callback($arguments, $options) {
     bee_render_text(array('value' => bt("Running installation in interactive mode. To run automatically (i.e. non-interactively), use the '--auto' option.\n")));
   }
 
-  // Get database connection string.
-  if (!empty($options['db'])) {
-    $db = $options['db'];
+  // Get database name.
+  if (!empty($options['db-name'])) {
+    $db_name = $options['db-name'];
+  }
+  else {
+    $db_name = rawurlencode(bee_check_plain(bee_input(bt('Database name'), '', TRUE)));
+  }
+  
+  // Get database username.
+  if (!empty($options['db-user'])) {
+    $db_username = $options['db-user'];
+  }
+  else {
+    $db_username = rawurlencode(bee_check_plain(bee_input(bt('Database username'), '', TRUE)));
+  }
+
+  // Get database password.
+  if (!empty($options['db-pass'])) {
+    $db_password = $options['db-pass'];
+  }
+  else {
+    $db_password = rawurlencode(bee_check_plain(bee_input(bt('Database password'), '', TRUE)));
+  }
+
+  // Get database host.
+  if (!empty($options['db-host'])) {
+    $db_host = $options['db-host'];
   }
   else {
-    $db_name = bee_check_plain(bee_input(bt('Database name'), '', TRUE));
-    $db_username = bee_check_plain(bee_input(bt('Database username'), '', TRUE));
-    $db_password = bee_check_plain(bee_input(bt('Database password'), '', TRUE));
     $db_host = bee_check_plain(bee_input(bt('Database host'), 'localhost', TRUE));
-    $db = "mysql://$db_username:$db_password@$db_host/$db_name";
   }
```

Comment:
With the host, I think it'd be nice to just assume it's `localhost` when the user provides a db-name (i.e. they've provided info via the command (meaning they probably don't want to be prompted), but they haven't provided the host so it's likely just the default).

Reply:
Done